be1-go/cli/mod.go

```diff
@@ -110,6 +161,59 @@ func Serve(cliCtx *cli.Context) error {
 	return nil
 }
 
+// serverConnectionLoop tries to connect to the remote servers following an exponential backoff strategy
+// it also listens for updates in the other-servers field and tries to connect to the new servers
+func serverConnectionLoop(h hub.Hub, wg *sync.WaitGroup, done chan struct{}, otherServers []string, updatedServersChan chan []string, connectedServers *map[string]bool) {
```

Comment:
This function needs a way to stop it when the server is closed, for example via a close channel.